src/site/content/en/blog/content-indexing-api/index.md

```diff
@@ -0,0 +1,363 @@
+as part of the `ServiceWorkerGlobalScope.`
+
+### Adding to the index {: #adding-items }
+
+Use the `add()` method to index URLs and their associated metadata. It's up to
+you to choose when items are added to the index. You might want to add to the
+index in response to an input, like clicking a "save offline" button. Or you
+might add items automatically each time cached data is updated via a mechanism
+like [periodic background
+sync](https://developers.google.com/web/updates/2019/08/periodic-background-sync).
+
+```js
+await registration.index.add({
+  // Required; set to something unique within your web app.
+  id: 'article-123',
+
+  // Required; this URL needs to be an offline-capable HTML page.
+  launchUrl: '/articles/123',
+
+  // Required; used in user-visible lists of content.
+  title: 'Article title',
+
+  // Required; used in user-visible lists of content.
+  description: 'Amazing article about things!',
+
+  // Required; used in user-visible lists of content.
+  icons: [{
+    src: '/img/article-123.png',
+    sizes: '64x64',
+    type: 'image/png',
+  }],
+
+  // Optional; valid categories are currently:
+  // 'homepage', 'article', 'video', 'audio', or '' (default).
+  category: 'article',
+});
+```
+
+Adding an entry only affects the content index; it does not add anything to the
+[Cache Storage
+API](https://developers.google.com/web/fundamentals/instant-and-offline/web-storage/cache-api).
+
+{% Aside 'note' %}
+During the call to `add()`, Chrome will make a request for
+each `icon`'s `src` URL to ensure that it has a copy of the icon to use when
+displaying a list of indexed content.
+
+If the call to `add()` happens in the `window` context (i.e. from your web
+page), this request will trigger a `fetch` event on your service worker.
+
+If `add()` is called within your service worker (perhaps inside another event
+handler), the request will **not** trigger the service worker's `fetch` handler.
+The icons will be fetched directly, without any service worker involvement. Keep
+this in mind if your icons rely on your `fetch` handler, perhaps because they
+only exist in the local cache and not on the network—if they do, make sure that
+you only call `add()` from the `window` context.
+{% endAside %}
+
+### Listing the index's contents {: #listing-items }
+
+The `getAll()` method returns a promise for an iterable list of indexed entries
+and their metadata. Returned entries will contain all of the data saved with
+`add()`.
+
+```js
+const entries = await registration.index.getAll();
+for (const entry of entries) {
+  // entry.id, entry.launchUrl, etc. are all exposed.
+}
+```
+
+### Removing items from the index {: #removing-items }
+
+To remove an item from the index, call `delete()` with the `id` of the item to
+remove:
+
+```js
+await registration.index.delete('article-123');
+```
+
+Removing an entry in this manner only affects the content index; it does not
```

Comment:
I think this can be a little more direct. 'Calling `delete()` only affects the index.'